packages/protocol/src/bridge/UMALayerZeroBridge.sol

```diff
@@ -0,0 +1,326 @@
+            _handleAssertTruthCmd(data);
+        } else {
+            revert("Invalid command type");
+        }
+    }
+
+    // Helper function to send LayerZero messages with quote
+    function _sendLayerZeroMessageWithQuote(
+        uint16 commandCode,
+        bytes memory commandPayload,
+        bool onlyQuote
+    )
+        internal
+        returns (MessagingReceipt memory receipt, MessagingFee memory fee)
+    {
+        bytes memory message = abi.encode(commandCode, commandPayload);
+
+        bytes memory options = OptionsBuilder
+            .newOptions()
+            .addExecutorLzReceiveOption(_getMaxExecutionGas(), 0);
+
+        // Get quote for the message
+        fee = _quote(
+            bridgeConfig.remoteChainId,
+            message,
+            options, // options
+            false // payInLzToken
+        );
+
+        if (onlyQuote) {
+            return (MessagingReceipt(0, 0, fee), fee);
+        }
+
+        // Check if contract has enough ETH
+        _requireSufficientETH(fee.nativeFee);
+
+        // Check gas thresholds and emit alerts before sending
+        _checkGasThresholds();
+
+        // Send the message using the external send function with ETH from contract
+        receipt = this._sendMessageWithETH{value: fee.nativeFee}(
+            bridgeConfig.remoteChainId,
+            message,
+            options,
+            fee
+        );
+
+        return (receipt, fee);
+    }
+
+    // External function to send LayerZero messages with ETH from contract balance
+    function _sendMessageWithETH(
+        uint32 _dstEid,
+        bytes memory _message,
+        bytes memory _options,
+        MessagingFee memory _fee
+    ) external payable returns (MessagingReceipt memory) {
+        require(msg.sender == address(this), "Only self-call allowed");
+        return
+            _lzSend(_dstEid, _message, _options, _fee, payable(address(this)));
+    }
+
+    function _handleAssertTruthCmd(bytes memory data) internal {
+        // Decode the data from the Market side (incoming data: assertionId, asserter, liveness, currency, bond, claim)
+        (
+            uint256 bridgeAssertionId,
+            address asserter,
+            uint64 liveness,
+            address bondTokenAddress,
+            uint256 bondAmount,
+            bytes memory claim
+        ) = data.decodeToUMAAssertTruth();
+
+        // TODO: Check if the assertionId is already in the mapping and send back an error message
+        // TODO: Check if the bond is enough and send back an error message
+
+        OptimisticOracleV3Interface optimisticOracleV3 = OptimisticOracleV3Interface(
+                optimisticOracleV3Address
+            );
+
+        IERC20 bondToken = IERC20(bondTokenAddress);
+
+        bondToken.approve(address(optimisticOracleV3), bondAmount);
+
+        bytes32 umaAssertionId = optimisticOracleV3.assertTruth(
+            claim,
+            asserter,
```

Comment:
shouldn't the bond be returned to this contract instead of the asserter?  and the asserter here is an eoa on converge?